pagopa/io-app
# feat: [IA-728] Saga and networking for upserting message attributes

## Short description
We implement the networking and logic for upserting message attributes _read_ and _archived_. I've connected the _read_ status to test it, while the _archived_ action is more complex due to sorting edge cases and will be handled in a separate task.

## List of changes proposed in this pull request
- new backend [endpoint](https://github.com/pagopa/io-app/pull/3848/files#diff-ac04d289df4b924fd0e2cee9ddaa414ad33332daf86424260ab533c21a284193R460)
- new [saga](ts/sagas/messages/watchUpsertMessageStatusAttribues.ts), [action](https://github.com/pagopa/io-app/pull/3848/files#diff-5b727d84caebb631cf6843fbd8f3595a97801341cee749b8d9535aabdbc12460R128), and [tests](ts/sagas/messages/__tests__/watchUpsertMessageStatusAttributes.test.ts)
- former [read](https://github.com/pagopa/io-app/pull/3848/files#diff-5b727d84caebb631cf6843fbd8f3595a97801341cee749b8d9535aabdbc12460R156) and [archived](https://github.com/pagopa/io-app/pull/3848/files#diff-5b727d84caebb631cf6843fbd8f3595a97801341cee749b8d9535aabdbc12460R165) actions are now deprecated
- _read_ and _archived_ are now [coming from backend](https://github.com/pagopa/io-app/pull/3848/files#diff-5d2069c099684d6a56a86df750b25c4cbf098b242c751215afab02b8f5a592a6R53) 🎉 
- had to fix [a small bug](https://github.com/pagopa/io-app/pull/3848/files#diff-bcbf6afc3d95f5b125321b43a35962fc8b5fa5186b29df65796d739bd9113817R35) to avoid migration of non-necessary messages, and [its test](https://github.com/pagopa/io-app/pull/3848/files#diff-67144522210873f6e25ec306e8dc26f17aa2b6626dd44f2c5a5b931db6ea53faR58) 
- the _read_ status is now handled at [MessageRouterScreen level](https://github.com/pagopa/io-app/pull/3848/files#diff-d26076117cc01e414c1a19c57e64ffbded662df3c965361935d414d9bb7fa01dR191) 
- the optimistic UI for _read_ status [is implemented](https://github.com/pagopa/io-app/pull/3848/files#diff-482523dcf54c84f4e40e09c47ee2eeca62c92423ef82ff98aae91839199d8c16R323)

## How to test

1. spin up the server and the app with pagination enabled, using [this branch](https://github.com/pagopa/io-dev-api-server/pull/143)
2. open an unread message
3. close it
4. if you uninstall the app and install again (keeping the server running!) you'll notice that the read status is persisted

https://user-images.githubusercontent.com/2094604/159440177-fcc50f5c-08ea-4945-a6f7-962fd261ad2b.mov
